src/gate_detect/README.md

```diff
@@ -0,0 +1,88 @@
+##Purpose
+
+This node takes in raw image data from a camera and determines whether there is a gate in the image, and if so what the 
+orientation of the gate is.
+
+The output of this node is a Gate construct that consists of 3 boolean values and 6 float values that are described as follows:
+
+
+          boolean detectedLeftPole:     False if left pole not seen, true if seen, 0 if not seen
+          
+          float angleLeftPole:          Angle from vertical centre of camera image to left pole
+
+          float distanceLeftPole:       Distance to left pole, 0 if not seen
+
+          boolean detectedRightPole:    False if right pole not seen, true if seen, 0 if not seen
+
+          float angleRightPole:         Angle from vertical centre of camera image to right pole
+
+          float distanceRightPole:      Distance to right pole, 0 if not seen
+
+          boolean detectedTopPole:      False if top pole not seen, true if seen
+
+          float angleTopPole:           Angle from horizontal centre of image to top pole, 0 if not seen
+
+          float distanceTopPole:        Distance to top pole, 0 if not seen
+
+
+##Method
+
+```
+
+       TOP VIEW
+       
+       GATE       ANGLE TO POLE
+                /        
+   POLE O    | \/  O POLE
+             |    /
+             |   / 
+             |  /  <--DISTANCE TO POLE
+             | /
+             |/          
+          --------
+          | ROBOT| 
+          |      |
+          --------
+```
+
+
+The gate detector used to process the image uses an interpolation function to estimate distance to the poles. Discrete data
+points must be collected of the pixel width of the poles of the gate at various distances. You then must use this data to 
+create an approximation function to map the pixel width of the poles to the distance from the pole.
+
+The current interpolation function is a power function: 
+
+(distance to pole) = interpolationConstant1 * (pixelWidthOfPole) ^ (interpolationConstant2)
```

Comment:
I'd probably remove this. In general we should make sure the external interface and broadly how this works are documented, but specific implementation details are likely to change, and this will rapidly become out of date.

Reply:
updated this comment